### ISSUE TITLE
day.ceil doesn’t always return boundary dates.

### ISSUE THREAD

**Issue by mbostock** [d3/d3-time#3]
Related #1 #2 mbostock/d3#1197.

```
TZ=Asia/Amman node
> d = new Date(2014, 2, 27)
Thu Mar 27 2014 00:00:00 GMT+0200 (EET)
> time.day.ceil(+d + 1)
Thu Mar 27 2014 23:00:00 GMT+0200 (EET)
```

The expected answer is `Fri Mar 28 2014 01:00:00 GMT+0300 (EEST)` The problem is that day.ceil relies on offseti, which increments date.setDate.

**Comment by mbostock** [comment]
This applies by extension to week and month.

**Comment by mbostock** [comment]
This appears to be fixed in Chrome and Firefox (per #1).
